### PR TITLE
ADU dataplane GA refresh

### PR DESCRIPTION
**Pull request by digimaun**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [x] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [x] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [x] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [x] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [x] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [x] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=6491&view=results

### CODE REVIEW

File: azext_iot/deviceupdate/commands_update.py [hunk 1]
```diff
@@ -181,6 +181,9 @@ def import_handler(lro: ARMPolling):
                 )
             elif lro.status() == "Failed":
                 try:
+                    logger.warning(
+                        "Cached contents (if any) from usage of --defer was not removed. "
```
Comment (avagraw) [inline azext_iot/deviceupdate/commands_update.py]: nit: Cached contents (if any) from usage of --defer **were** not removed.

File: azext_iot/deviceupdate/params.py [hunk 2]
```diff
@@ -305,7 +325,13 @@ def load_deviceupdate_arguments(self, _):
         )
 
     with self.argument_context("iot device-update device deployment") as context:
-        context.argument("deployment_id", options_list=["--deployment-id", "--did"], help="Deployment Id.")
+        context.argument(
+            "deployment_id",
+            options_list=["--deployment-id", "--did"],
+            help="The caller-provided deployment Id. This cannot be longer than 73 characters, "
+            "must be all lower-case, and cannot contain '&', '^', '[', ']', '{', '}', '|', '<', '>', "
```
Comment (avagraw) [inline azext_iot/deviceupdate/params.py:332]: Is this list of characters taken from the official API documentation? Are all the other special characters supported?

File: azext_iot/tests/deviceupdate/conftest.py [hunk 3]
```diff
@@ -22,8 +22,10 @@
 
 ACCOUNT_RG = settings.env.azext_iot_testrg
 VALID_IDENTITY_MAP = {"system": 1, "user": 1}
-DEFAULT_ADU_RBAC_SLEEP_SEC = 90
-ADU_CLIENT_DTMI = "dtmi:azure:iot:deviceUpdateModel;1"
+DEFAULT_ADU_RBAC_SLEEP_SEC = 120
```
Comment (avagraw) [inline azext_iot/tests/deviceupdate/conftest.py]: Were we seeing RBAC failures with sleep time of 90 seconds? Or was there any other driver for this change?

File: azext_iot/deviceupdate/params.py (hunk 2 above)
Comment (digimaun) [reply azext_iot/deviceupdate/params.py:332]: Yah that is right from the official documentation.

Comment (digimaun) [reply azext_iot/deviceupdate/params.py:332]: Honestly I haven't tried every other special character not in that list, but I'd like to think API documentation would be accurate and if something was missed theoretically the API should error (and with luck we get an issue in the repo requesting doc improvement). In later refreshes we could spot check for additional API doc updates.

File: azext_iot/tests/deviceupdate/conftest.py (hunk 3 above)
Comment (digimaun) [reply azext_iot/tests/deviceupdate/conftest.py]: Very little...ADU tests have been pretty stable, but there was a timing issue in this last iteration which motivated the increase. I'll reduce it by half and keep an eye on it.

**Review by avagraw — APPROVED** [review]

**Pull request merged**